client/locations-list-map.js

```diff
@@ -1146,6 +1171,65 @@ function loadMapScript(searchParams, data, filters) {
   document.head.appendChild(scriptTag);
 }
 
+function initContactModal() {
+  let lastOrg = null;
+  $('#contactModal').on('show.bs.modal', function (event) {
+    const el = $(event.relatedTarget);
+    const email = el.data('email');
+    const name = el.data('name');
+    const modal = $(this);
+
+    if(lastOrg != name) {
+      lastOrg = name;
+      $('#sender-name').val(null);
+      $('#sender-email').val(null);
+      $('#message-subject').val(null);
+      $('#message-text').val(null);
+    }
+
+    modal.find('.modal-title').text($.i18n('ftm-email-form-title-label') + ' ' + name);
+    modal.find('#message-recipient').val(email);
+  });
+
+  $('#contactModal #send-message').on('click', function (event) {
+    $('.contact-error').html('&nbsp;');
+    $('#send-message').prop('disabled', true);
+
+    $.post(
+      'https://maskmailer.herokuapp.com/send',
+      {
+        name: $('#sender-name').val(),
+        from: $('#sender-email').val(),
+        subject: $('#message-subject').val(),
+        text: $('#message-text').val(),
+        introduction: $.i18n('ftm-email-introduction'),
+        to: $('#message-recipient').val(),
+        'g-recaptcha-response': grecaptcha.getResponse(),
+      }
+    ).done(function(result) {
+      $('.contact-form').css('display', 'none');
+      $('.contact-success').css('display', 'block');
+      $('#send-message').prop('disabled', false);
+      grecaptcha.reset();
+
+      setTimeout(function() {
+        $('#contactModal').modal('hide');
+      }, 5000);
```

Comment:
why is this on a timeout?

Reply:
It replaces the modal content with a "your email has been sent" confirmation. But there's no need to leave that up indefinitely.